### PR TITLE
Started only loading gles3 functions if we have a gles3 context

### PR DESCRIPTION
**Pull request by gaaclarke**
test: exists in https://github.com/flutter/engine/pull/56596
fixes: https://github.com/flutter/flutter/issues/158995

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: impeller/renderer/backend/gles/proc_table_gles.cc [hunk 1]
```diff
@@ -121,7 +121,16 @@ ProcTableGLES::ProcTableGLES(  // NOLINT(google-readability-function-size)
         reinterpret_cast<decltype(proc_ivar.function)>(fn_ptr); \
     proc_ivar.error_fn = error_fn;                              \
   }
-  FOR_EACH_IMPELLER_GLES3_PROC(IMPELLER_PROC);
+
+  GLint major_version = 0;
```
Comment (jonahwilliams) [inline impeller/renderer/backend/gles/proc_table_gles.cc:125]: Use `description_->GetGlVersion().IsAtLeast(...)`

Comment (gaaclarke) [reply impeller/renderer/backend/gles/proc_table_gles.cc]: done

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Comment by gaaclarke** [comment]
I'm going to wait to land this until https://github.com/flutter/engine/pull/56596 does, since it has a good test to make sure this doesn't break anything.

**Comment by gaaclarke** [comment]
I should have rebased before landing, oh well I tested it locally and is almost surely good.

**Pull request merged**